src/cljs/rems/flash_message.cljs

```diff
@@ -40,7 +47,9 @@
 
     :component-will-unmount
     (fn [_this]
-      (rf/dispatch [::reset]))
+      (let [message @(rf/subscribe [::message])]
+        (when (expired? message)
```

Comment:
I tried to check when component-will-unmount is called, but couldn't easily think of a case in which this check is useful to have. Can someone clarify?

Reply:
at least in dev mode when doing a live reload the components will be un- and remounted because there's a rerender

something similar might happen when an application is reloaded

Reply:
Hmm, ok. I guess it doesn't hurt to have the message expired and not re-shown in these kind of cases, so it's fine.

Reply:
It happens when the page is changed as a response to a successful submit, so that the message will be shown on the next page since the user did not have time to see it on the original page. And when the user navigates manually to another page, the message will have expired and it won’t be shown anymore.